src/clang-c-frontend/clang_c_adjust_code.cpp

```diff
@@ -42,6 +42,23 @@ void clang_c_adjust::adjust_code(codet &code)
     adjust_decl_block(code);
   else
   {
+    if (statement == "expression" && is_array_like(code.op0().type()))
+    {
+      /* An array-type'd statement like "y->ss;" where y is a pointer to
+       *
+       *   struct { int ss[128]; }
+       *
+       * is not assumed to not exist by the dereference code. Thus, convert it
```

Comment:
Is this double negation intentional?

Reply:
It's not. ;) Thanks, I'll fix the comment.